### PR TITLE
Update context docs

### PR DESCRIPTION
**Pull request by andrzejewsky**
Update docs

### CODE REVIEW

File: packages/core/docs/general/context.md [hunk 1]
```diff
@@ -12,137 +18,159 @@ $vsf {
...
+The `tag` in the configuration is sort of identifier what will be used as a context key.
 
-  // $ct.api
-  // $ct.client
-  // $ct.config
-}
-```
+Inside of `onSetup` function you can put a creation of the connection or anything else that's needed for preparing an api connection - remember, onSetup must always return the `client` and `config`.
+
+## Api client usage
```
Comment (filrak) [inline packages/core/docs/general/context.md:70]: missing info (tip) like: 'If you're using our default setup and Nuxt you can omit this part" or similar

File: packages/core/docs/general/context.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
 # Context API
```
Comment (filrak) [inline packages/core/docs/general/context.md:1]: imho this is 100% content for either integrations category od a new one like "Internals"

File: packages/core/docs/integrate/integration-guide.md [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+  return productResponse.data
+}
+```
+
+Each api function always contains `context` as a first parameter. This is the place where you always have access to the client and config of your api connection. Obviously, during the using api client, you will be using these functions without thinking about the context - the vsf core handles this. At the end, you need to provide that function to the api creation logic (section above)
+
+## 2. Creating composables
+
+Composables are the major part of integration. That exactly the place where the business logic comes in. We always serving this package as a integration along with corresponding nuxt module.
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:174]:
```suggestion
Composables are a major part of the integration. That exactly the place where the business logic comes in. We always serve this package as integration along with the corresponding Nuxt module.
```

File: packages/core/docs/integrate/integration-guide.md [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+
+### 2.1 Exposing integration plugin wrapper
+
+The first thing that you do, is exposing a nuxt plugin wrapper for creation and configuring your integration by nuxt environment. Within your composables package you have to add these few lines:
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:178]: Let's make a convention about naming libraries with capitalized first letter (except webpack and npm 😅 )
```suggestion
The first thing that you do, is exposing a Nuxt plugin wrapper for the creation and configuring your integration by Nuxt environment. Within your composables package you have to add these few lines:
```

File: packages/core/docs/integrate/integration-guide.md [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+
+That will expose the integration plugin wrapper, so the nuxt will be able to create your integration within the context or extend already existing one, using a plugin
+
+### 2.2 Creating nuxt module
+
+Inside of the composables packages you have to create another directory, next to `src` called `nuxt`. In that directory we need to place our nuxt module. The nuxt module is taking care of anything you want during the integration to be launched: adding plugins, injecting into build process, creating some aliases and more. In the basic implementation of that module will add just a plugin that will configure our application (using a wrapper that you have already exposed)
+
+Example of plugin
+```js
+// coposables/nuxt/plugin.js
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:196]:
```suggestion
// composables/nuxt/plugin.js
```

File: packages/core/docs/integrate/integration-guide.md [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+export default integrationPlugin(({ app, integration }) => {
+  const settings = { api: '/graphql', user: 'root' }
+
+  integration.configure({ ...moduleOptions, ...settings })
+});
+```
+
+The module:
+```js
+// coposables/nuxt/index.js
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:210]:
```suggestion
// composables/nuxt/index.js
```

File: packages/core/docs/integrate/integration-guide.md [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+  },
+};
+
+const useCart = useCartFactory(factoryParams);
+
+export default useCart;
+```
+
+```ts
+// coposables/src/index.js
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:260]:
```suggestion
// composables/src/index.js
```

File: packages/core/docs/integrate/integration-guide.md [hunk 8]
```diff
@@ -0,0 +1,369 @@
...
+export { useCart }
+```
+
+Each function inside of factory params has the context in the very first argument. The second argument contains always the given parameters to the function (eg. product data in adding to cart function)
+
+### 2.3b Creating composables without our factories
+
+Sometimes there is a need when you want to omit using our factories. You can have a really complex integration or you want to follow totally your way of implementing this. In this case, you can still stay stick to the Vue Storefront architecture and create your own composable, by creating your own factory, or even ordinary composable by yourself.
+
+However, you have to keep in mind that you need to handle context and reactive properties as well - Of course you have helpers for that.
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:272]:
```suggestion
However, you have to keep in mind that you need to handle context and reactive properties as well - of course you have helpers for that.
```

File: packages/core/docs/integrate/integration-guide.md [hunk 9]
```diff
@@ -0,0 +1,369 @@
...
+
+```ts
+// coposables/src/useCart/index.js
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:275]:
```suggestion
// composables/src/useCart/index.js
```

File: packages/core/docs/integrate/integration-guide.md [hunk 10]
```diff
@@ -0,0 +1,369 @@
...
+    return context.$ownAPI.updateCart(product)
+  }
+
+  return { addToCart, cart }
+}
+```
+
+## 3. Getters
+
+Our composables always return the properties:
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:292]:
```suggestion
Our composables always return the following properties:
```

File: packages/core/docs/integrate/integration-guide.md [hunk 11]
```diff
@@ -0,0 +1,369 @@
...
+
+- a field that contains the response - always when you load or modify the state of the given feature, we generate a new response which is being returned by composable. It always has a proper name that refers to the its purpose eg. cart, products, category etc.
```
Comment (NataliaTepluhina) [inline packages/core/docs/integrate/integration-guide.md:294]:
```suggestion
- a field that contains the response - when you load or modify the state of the given feature, we generate a new response that is being returned by composable. It always has a proper name that refers to its purpose eg. cart, products, category etc.
```

**Review by NataliaTepluhina — COMMENTED** [review]
I've left a handful of comments here and there, would be happy to discuss or explain any point there

**Pull request merged**